js/setup.js

```diff
@@ -39,12 +39,114 @@ var WIZARD_EYES_COLORS = [
   'green'
 ];
 
+var WIZARD_FIREBALL_COLORS = [
+  '#ee4830',
+  '#30a8ee',
+  '#5ce6c0',
+  '#e848d5',
+  '#e6e848'
+];
+
 var TOTAL_WIZARDS = 4;
 var wizards = [];
 
 var similarList = document.querySelector('.setup-similar-list');
 var template = document.querySelector('#similar-wizard-template').content.querySelector('.setup-similar-item');
 
+var ESC_KEY = 'Escape';
+var ENTER_KEY = 'Enter';
+
+var MIN_NAME_LENGTH = 2;
+var MAX_NAME_LENGTH = 25;
+
+var setup = document.querySelector('.setup');
+var setupOpen = document.querySelector('.setup-open');
+var setupClose = setup.querySelector('.setup-close');
+
+var userNameInput = setup.querySelector('.setup-user-name');
+var coatColorInput = setup.querySelector('input[name=coat-color]');
+var eyesColorInput = setup.querySelector('input[name=eyes-color]');
+var fireballColorInput = setup.querySelector('input[name=fireball-color]');
+
+var wizardCoat = setup.querySelector('.wizard-coat');
+var wizardEyes = setup.querySelector('.wizard-eyes');
+var wizardFireball = setup.querySelector('.setup-fireball-wrap');
+
+// при клике на esc вызывается ф-ция закрытия окна
+var closePopupOnEsc = function (evt) {
+  if (evt.key === ESC_KEY && document.activeElement !== userNameInput) {
+    closePopup(setup);
+  }
+};
+
+// закрытие окна
+var closePopup = function () {
+  setup.classList.add('hidden');
+  document.removeEventListener('keydown', closePopupOnEsc);
```

Comment:
```suggestion
  document.removeEventListener('keydown', documentKeydownHandler);
```
Есть два способа, или писать функции непосредственно на месте, но тогда их нельзя будет отключить от слушания, или придумать им всем названия, как советует академия в https://up.htmlacademy.ru/javascript/19/criteries#d4, и уже внутри хендлеров вызывать непосредственно функции, которые делают что-то